### PR TITLE
Package ESMValTool

### PR DESCRIPTION
**Pull request by zklaus**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 1 to supersede the earlier build of https://github.com/conda-forge/esmvaltool-python-feedstock.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by zklaus** [comment]
tl;dr: Can rename conda-forge/esmvaltool-python-feedstock to esmvaltool-feedstock?


@conda-forge/staged-recipes, I would like to request some guidance on how to move on. The situation is as follows:

The project is principally a Python package for the analysis of climate models. It supports user written diagnostics in several languages, at the moment Python, R, NCL, and Julia. The packaging is now realized as a multi-output where `esmvaltool-python` is the main package containing both the tool itself and the Python diagnostics, separate packages for the R and NCL diagnostics, called `esmvaltool-r` and `esmvaltool-ncl`, respectively, and an overall meta package, `esmvaltool`, that depends on all of these.
In other words, `esmvaltool` depends on `esmvaltool-python`, `esmvaltool-r`, and `esmvaltool-ncl`; `esmvaltool-r` depends on `esmvaltool-python`; `esmvaltool-ncl` depends on `esmvaltool-python`.

Because that is the principal package, we started with the esmvaltool-python-feedstock, with the intention of adding esmvaltool-r-feedstock, esmvaltool-ncl-feedstock, and esmvaltool-feedstock.
However, with the successful integration of the multi-output recipe in esmvaltool-python-feedstock (onda-forge/esmvaltool-python-feedstock#6), the other feedstocks are no longer necessary, so ideally we would rename esmvaltool-python-feedstock to esmvaltool-feedstock. Is this possible?

**Comment by zklaus** [comment]
After a discussion on gitter with @tdegeus and @beckermr, we decided to continue with this PR for a single esmvaltool-feedstock with a multi-output build as developed in conda-forge/esmvaltool-python-feedstock#6 and subsequent retirement of esmvaltool-python-feedstock.

That is why the build-number is 1, not 0, because it replaces the earlier esmvaltool-python build. This is also the reason why the linter complains about an existing package with the same name.

**Comment by zklaus** [comment]
This is a noarch package that is failing on windows due to a missing dependency. This should not stop us from merging, c.f. #14700.

**Comment by zklaus** [comment]
@conda-forge/esmvaltool-python, could you guys please have a look? This is the same multi-output recipe that we have been developing in conda-forge/esmvaltool-python-feedstock#6, only in a new feedstock (esmvaltool-feedstock), plus the following changes:

- I removed most of the pins; now the pinning is as in upstream
- I added basic testing for the top-level (`esmvaltool`) package
- I added the `console_scripts` entry points to comply with [the noarch requirements](https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-python)
- Other minor changes (formatting etc)

If you agree, I think this can go in. The Linux build works, though both osx and windows build here fail due to missing or unresolvable dependencies. Note that this does not prevent one form trying to install it on osx or windows since it will be built as noarch. Then we can try to improve the situation by working on the missing dependencies on those platforms.

**Comment by valeriupredoi** [comment]
@zklaus very many thanks for battling and winning the fight with our dependencies list and making sure this works! I am very happy to see this through (gonna approve after I post this message), since the linux-64 distribution works fine, all's good by me. Note, however, that we may get a kick in build times and (probably) better deps solving by retiring Python 3.7, our tests have already started failing badly at figuring an env with it on [GA](https://github.com/ESMValGroup/ESMValTool/runs/3267056098?check_suite_focus=true). Cheers, mate :beer:

**Review by valeriupredoi — APPROVED** [review]

File: recipes/esmvaltool/yum_requirements.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+xorg-x11-server-Xorg
```
Comment (bouweandela) [inline recipes/esmvaltool/yum_requirements.txt:1]: Why do we need x11? I'm pretty sure our users are running the tool on systems without a graphical user interface.

**Comment by bouweandela** [comment]
It looks like the `esmvaltool-python` package is building fine for OSX, but failing to solve the dependencies for `esmvaltool-ncl`. In our documentation we say that `esmvaltool-python` can be installed on OSX, see [here](https://docs.esmvaltool.org/en/latest/quickstart/installation.html#installation-on-macosx). Would it be possible to build just the `esmvaltool-python` package for OSX now, so that remains available for our OSX users?

File: recipes/esmvaltool/yum_requirements.txt (hunk 1 above)
Comment (zklaus) [reply recipes/esmvaltool/yum_requirements.txt:1]: Removing x11 dependencies is exactly why we need this. It came in as part of conda-forge/esmvaltool-python-feedstock#1 when we moved from depending on `matplotlib` to `matplotlib-base` which removes the `pyqt` dependency and a whole bunch of x11 dependencies. To still be able to build, we need this yum requirement on the build machine.

**Comment by zklaus** [comment]
> It looks like the `esmvaltool-python` package is building fine for OSX, but failing to solve the dependencies for `esmvaltool-ncl`. In our documentation we say that `esmvaltool-python` can be installed on OSX, see [here](https://docs.esmvaltool.org/en/latest/quickstart/installation.html#installation-on-macosx). Would it be possible to build just the `esmvaltool-python` package for OSX now, so that remains available for our OSX users?

Everything is marked as `noarch` which means that once this PR is merged and esmvaltool has its own feedstock, things will be build only once (on a Linux machine), but will be installable on any of the platforms. Of course, where missing dependencies prevent a successful installation, the installation will then fail, but installing `esmvaltool-python` on osx should not be a problem.

**Comment by bouweandela** [comment]
Thanks @zklaus, let's get this merged!

**Comment by zklaus** [comment]
@dopplershift, since you already helped with #14700 and #14400, would you mind having a look? This PR is in part designed to replace #14700. Sorry for the confusion.

**Comment by beckermr** [comment]
All of the maintainers in the recipe need to comment that they are ok being on it or they need to be removed.

**Comment by zklaus** [comment]
@valeriupredoi, @nielsdrost, @jvegasbsc, @bouweandela could you guys please comment here, confirming that you are willing to co-maintain this package?

**Comment by bouweandela** [comment]
I'm OK with being on it, thanks for adding me.

**Comment by nielsdrost** [comment]
Yes, no problem for being on the list of co-maintainers, thanks for the effort!

**Comment by valeriupredoi** [comment]
cheers for adding me @zklaus and of course (my horse) I am OK with being a maintainer. Sorry for the delayed reply, I was away :+1: :beer:

**Comment by zklaus** [comment]
@beckermr, thanks for your help. I have removed one maintainer and gotten the comments from all the remaining ones (see above).
Do you have any other comments?

**Comment by zklaus** [comment]
Thanks, @beckermr :tada: !

**Comment by zklaus** [comment]
> I'm OK with being on it, thanks for adding me. And sorry for the late reply

Thanks Javi! Unfortunately, it's too late for this PR, but we will add you in an update on the new feedstock.
